lib/AST/ASTImporter.cpp

```diff
@@ -931,8 +931,30 @@ bool ASTNodeImporter::ImportDeclParts(NamedDecl *D, DeclContext *&DC,
                                       DeclarationName &Name, 
                                       NamedDecl *&ToD,
                                       SourceLocation &Loc) {
+  // Check if RecordDecl is in FunctionDecl parameters to avoid infinite loop.
+  // example: int struct_in_proto(struct data_t{int a;int b;} *d);
+  DeclContext *OrigDC = D->getDeclContext();
+  FunctionDecl *FunDecl;
+  if (isa<RecordDecl>(D) && (FunDecl = dyn_cast<FunctionDecl>(OrigDC))) {
+    bool DoNotImport = !FunDecl->hasBody();
+    if (!DoNotImport) {
```

Comment:
What if we only see the prototype but it does not have a body? I think that is a perfectly valid case. Do we want to give a warning for that case?

Reply:
If we only have a prototype and we still have a recorddecl and this functiondecl is the declcontext, than it is the same thing and we want to give a warning as we can't import this the same way. If the importer starts importing it, the same loop we are in.